test/cloudformation/cloudformation_test.go

```diff
@@ -0,0 +1,177 @@
+package cloudformation
+
+import (
+	"context"
+	"flag"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	uuid2 "github.com/google/uuid"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	timeOut       = 15
+	instanceIdKey = "InstanceId"
+	namespace     = "CFMetricTest"
+)
+
+var (
+	dimensionProviders = []dimension.IProvider{
+		&dimension.CustomDimensionProvider{Provider: dimension.Provider{}},
+	}
+	dimensionFactory = dimension.Factory{Providers: dimensionProviders}
+	packagePath      = flag.String("package_path", "", "path to download package")
+	iamRole          = flag.String("iam_role", "", "iam role for cf ec2 instance")
+	keyName          = flag.String("key_name", "", "key name for ec2 instance")
+	metricName       = flag.String("metric_name", "", "metric to look for")
+)
+
+func TestCloudformation(t *testing.T) {
+	log.Printf("Package path %s iam role %s key name %s metric name %s", *packagePath, *iamRole, *keyName, *metricName)
+	cxt := context.Background()
+	stackName := createStackName()
+
+	client := createCloudwatchClient(cxt)
+
+	startStack(cxt, stackName, client)
+
+	instanceId := findStackInstanceId(cxt, stackName, client)
+
+	for i := 0; i < timeOut; i++ {
+
+	}
+	// Sleep so metrics can be added to cloudwatch
+	log.Printf("Sleep for one minute to collect metrics")
+	time.Sleep(time.Minute)
+
+	dims, failed := dimensionFactory.GetDimensions([]dimension.Instruction{
+		{
+			Key:   instanceIdKey,
+			Value: dimension.ExpectedDimensionValue{Value: aws.String(instanceId)},
+		},
+	})
+
+	if len(failed) > 0 {
+		deleteStack(cxt, stackName, client)
+		t.Fatalf("Failed to generate dimensions")
+		return
+	}
+
+	fetcher := metric.MetricValueFetcher{}
+	values, err := fetcher.Fetch(namespace, *metricName, dims, metric.AVERAGE, test_runner.HighResolutionStatPeriod)
```

Comment:
It is possible another template wants to search for a different metric. Such as for windows. As of now I don't think we need to add other os since we test in other integration tests.